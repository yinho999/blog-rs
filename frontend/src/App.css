body {
  background-color: rgb(28, 30, 31); /* Replace with your desired color */
}