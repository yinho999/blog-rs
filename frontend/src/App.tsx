import "./App.css";
import AppRouter from "./router";
import { initialState, reducer } from "./reducer";
import { useReducer } from "react";
import { AppContext } from "./context/AppContext";
import "@mui/material/styles";
// import Header from "./components/Header";
const App = () => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <>
      {/*<Header />*/}
      <AppContext.Provider value={{ state, dispatch }}>
        <AppRouter />
      </AppContext.Provider>
    </>
  );
};

export default App;
