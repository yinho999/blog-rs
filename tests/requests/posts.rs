use insta::{assert_debug_snapshot, with_settings};
use blog::app::App;
use loco_rs::testing;
use serial_test::serial;
use super::prepare_data;

macro_rules! configure_insta {
    ($($expr:expr),*) => {
        let mut settings = insta::Settings::clone_current();
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_suffix("post_request");
        let _guard = settings.bind_to_scope();
    };
}

#[tokio::test]
#[serial]
async fn can_create_post() {
    configure_insta!();

    testing::request::<App, _, _>(|request, ctx| async move {
        let user = prepare_data::init_user_login(&request, &ctx).await;

        let (auth_key, auth_value) = prepare_data::auth_header(&user.token);
        let payload = serde_json::json!({
            "title": "loco",
            "description": "loco post test description",
            "content": "loco post test",
        });

        let add_post_request = request
            .post("/api/posts")
            .add_header(auth_key, auth_value)
            .json(&payload)
            .await;

        with_settings!({
            filters => {
                 let mut combined_filters = testing::CLEANUP_DATE.to_vec();
                    combined_filters.extend(vec![
                    (r#"\"id\\":\d+"#, r#""id\":ID"#)
                    , (r#"\"user_id\\":\d+"#, r#""user_id\":USER_ID"#)
                ]);
                    combined_filters
            }
        }, {
            assert_debug_snapshot!(
            (add_post_request.status_code(), add_post_request.text())
        );
        });
    })
        .await;
}