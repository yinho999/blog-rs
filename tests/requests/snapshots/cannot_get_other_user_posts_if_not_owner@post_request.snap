---
source: tests/requests/posts.rs
expression: "(posts.status_code(), posts.text())"
---
(
    200,
    "[{\"id\":ID,\"title\":\"loco3\",\"description\":\"loco post test description3\",\"content\":\"loco post test3\",\"created_at\":\"DATE\",\"updated_at\":\"DATE\"}]",
)
